src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -820,41 +820,46 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
             }
         }
 
-        private static TypedConstant GetParamArrayArgument(ParameterSymbol parameter, ImmutableArray<TypedConstant> constructorArgsArray, int argumentsCount, int argsConsumedCount, Conversions conversions)
+        private static TypedConstant GetParamArrayArgument(ParameterSymbol parameter, ImmutableArray<TypedConstant> constructorArgsArray,
+            ImmutableArray<string> constructorArgumentNamesOpt, int argumentsCount, int argsConsumedCount, Conversions conversions)
         {
             Debug.Assert(argsConsumedCount <= argumentsCount);
 
+            // If there's a named argument, we'll use that
+            if (!constructorArgumentNamesOpt.IsDefault)
+            {
+                int argIndex = constructorArgumentNamesOpt.IndexOf(parameter.Name);
+                if (argIndex >= 0)
+                {
+                    if (TryGetNormalParamValue(parameter, constructorArgsArray, argIndex, conversions, out var namedValue))
+                    {
+                        return namedValue;
+                    }
+
+                    // A named argument for a params parameter must be trailing, so expanded params must have a single value
+                    return new TypedConstant(parameter.Type, ImmutableArray.Create(constructorArgsArray[argIndex] ));
+                }
+            }
+
             int paramArrayArgCount = argumentsCount - argsConsumedCount;
+
+            // If there are zero arguments left
             if (paramArrayArgCount == 0)
             {
                 return new TypedConstant(parameter.Type, ImmutableArray<TypedConstant>.Empty);
             }
```

Comment:
Minor point: consider handling the `zero arguments left` case first.

Reply:
No, I think it must be after the named arguments case.

---
In reply to: [140549685](https://github.com/dotnet/roslyn/pull/22231#discussion_r140549685) [](ancestors = 140549685)